pulsar-io/azure-data-explorer/src/test/java/org/apache/pulsar/io/azuredataexplorer/ADXSinkE2ETest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.azuredataexplorer;
+
+import com.microsoft.azure.kusto.data.Client;
+import com.microsoft.azure.kusto.data.ClientFactory;
+import com.microsoft.azure.kusto.data.KustoOperationResult;
+import com.microsoft.azure.kusto.data.KustoResultSetTable;
+import com.microsoft.azure.kusto.data.auth.ConnectionStringBuilder;
+import org.apache.pulsar.functions.api.Record;
+import org.apache.pulsar.functions.instance.SinkRecord;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class ADXSinkE2ETest {
+
+    private String table = "ADXPulsarTest_" + ThreadLocalRandom.current().nextInt(0, 100);
+
+    private Client kustoAdminClient = null;
+    String database;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        Objects.requireNonNull(System.getenv("kustoDatabase"), "kustoDatabase not set.");
+        Objects.requireNonNull(System.getenv("kustoCluster"), "kustoCluster not set.");
```

Comment:
String appId = Objects.requireNonNull(System.getenv("kustoAadAppId"), "kustoAadAppId not set.");
String database = Objects.requireNonNull(System.getenv("kustoDatabase"));

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-